src/main.rs

```diff
@@ -134,6 +143,36 @@ impl Expr {
     }
 }
 
+impl Func {
+    fn eval(&self, arg: Value) -> Value {
+        match self {
+            Func::Expr(e) => e.eval(&[]),
+            Func::Fst => match arg {
+                Value::Pair(fst, _snd) => *fst,
+                _ => panic!("Fst: expected (_, _)"),
+            },
+            Func::Snd => match arg {
+                Value::Pair(_fst, snd) => *snd,
+                _ => panic!("Snd: expected (_, _)"),
+            },
+            Func::U16Be => match arg {
+                Value::Pair(fst, snd) => match (fst.as_ref(), snd.as_ref()) {
+                    (Value::U8(hi), Value::U8(lo)) => Value::U16(u16::from_be_bytes([*hi, *lo])),
+                    (_, _) => panic!("expected (U8, U8)"),
+                },
+                _ => panic!("U16Be: expected (_, _)"),
+            },
+            Func::Stream => match arg {
+                Value::Seq(vs) => {
+                    // FIXME could also condense nested sequences
+                    Value::Seq(vs.into_iter().filter(|v| *v != Value::Unit).collect())
+                }
+                _ => panic!("Stream: expected Seq"),
+            },
```

Comment:
Mildly terrified about what type this would have 😅

Reply:
Yes that's one advantage of not needing to typecheck this yet :nerd_face: 

But I think with further experience we will probably refine these primitives into something more tractable. (An easier way to type it would just make it condense [[T]] into [T]).